packages/next-app/src/contexts/AdminToolsContext.tsx

```diff
@@ -55,15 +67,43 @@ interface AdminToolsContextType {
   handleSetAction: (action: ActionAttribute) => void;
   submit: boolean;
   handleSubmit: Dispatch<SetStateAction<boolean>>;
+  selectedFilters: { [key: string]: string };
+  changeSelectedFilters: (key: string, value: string) => void;
+  clearSelectedFilter: (field: string) => void;
 }
 
 export const AdminToolsContext = createContext({} as AdminToolsContextType);
 
 export function AdminToolsProvider({ children }: { children: ReactNode }) {
+  const filterInitialState = {
+    operationResponsible: "",
+  };
+
   const [selectedAction, setSelectedAction] =
     useState<ActionAttribute>(initialState);
+
   const [submit, handleSubmit] = useState<boolean>(false);
 
+  const [selectedFilters, setSelectedFilters] = useState(filterInitialState);
+
+  function changeSelectedFilters(key: string, value: string) {
+    setSelectedFilters((prevState) => {
+      return {
+        ...prevState,
+        [key]: value,
+      };
+    });
+  }
+
+  function clearSelectedFilter(field: string) {
+    setSelectedFilters((prevState) => {
+      return {
+        ...prevState,
+        [field]: "",
+      };
+    });
+  }
+
```

Comment:
`#could` make these functions API more clear. I think `key` in `changeSelectedFilters` and `field` in `clearSelectedFilters` refer to the same thing, right? So they should probably have the same name. Also, you could leverage Typescript to qualify  a bit more what the allowed filterKeys are and instead of saying that they're a string you can type these params with the allowed filterKeys from a preset of values using e.g. an Enum

Reply:
`#could `part is done, thinking on the allowed filterKeys part